.github/workflows/pypi_publish.yml

```diff
@@ -48,8 +56,104 @@ jobs:
           name: wheels-sdist
           path: ./wheelhouse/*.tar.gz
 
+  build-wheels:
+    name: 'Wheels: ${{ matrix.os }} ${{ matrix.arch }} ${{ matrix.python-version }}'
+    runs-on: ${{ matrix.os }}
+    continue-on-error: true
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-latest, macos-latest]
+        python-version: ['3.12']
+        include:
+          - os: ubuntu-latest
+            arch: x86_64
+          # - os: windows-latest
+          #   arch: AMD64
+          - os: macos-latest
+            arch: arm64
+
+    steps:
+    - name: Check out the repo
+      uses: actions/checkout@v4
+      with:
+        fetch-depth: 0  # get the non-merge commit for PRs
+
+    # Install openmpi on macos
+    - name: Install openmpi and hdf5 on macos
+      if: matrix.os == 'macos-latest'
+      run: |
+        brew install open-mpi
+        echo "PKG_CONFIG_PATH=$(brew --prefix)/lib/pkgconfig:$(brew --prefix)/opt/open-mpi/lib/pkgconfig:$PKG_CONFIG_PATH" >> $GITHUB_ENV
+        echo "CPATH=$(brew --prefix)/include:$CPATH" >> $GITHUB_ENV
+        echo "LIBRARY_PATH=$(brew --prefix)/lib:$LIBRARY_PATH" >> $GITHUB_ENV
+
+    # Set up python on ubuntu (build runs in containers, so can't use conda)
+    # Can't use conda on macos, the binaries it installs are invalid according
+    # to meson: "executables created by cpp compiler are not runnable"
```

Comment:
Did you try installing the conda-provided compiler tools? (`clangxx_osx-64` on mac and `gxx_linux-64` on linux) 
https://docs.conda.io/projects/conda-build/en/stable/resources/compiler-tools.html

Reply:
Thanks for the information - as you mention further down importing packages outside of conda is messy, so I abandoned conda entirely for wheel building.